source/configuration_manual/zlib_plugin.rst

```diff
@@ -33,11 +33,14 @@ The ``zlib_save`` setting selects the compression algorithm (currently
 supported values are: gz, bz2, lz4, zstd) to use when saving a new mail.
 The ``zlib_save_level`` setting sets the compression level used.
 
-.. note::
-   Currently, the compression level must be an integer in the range 1 to 9
-   regardless of the algorithm selected.  The default level is 6.  These
-   values may not be sensical with compression algorithms other than gz and
-   bz2.
+.. versionchanged:: 2.3.15
+
+You can use per-algorithm compression levels, and defaults. Prior to v2.3.15,
+the compression level must be an integer in the range 1 to 9 regardless of the
```

Comment:
Would this look ok if the text was together with the versionchanged? I mean..:

```
.. versionchanged:: 2.3.15 You can use per-algorithm compression levels, and defaults. Prior to v2.3.15,
   the compression level must be an integer in the range 1 to 9 regardless of the
   ...
```

The benefit is that it clearly indicates what changed in 2.3.15. Now it's not so obvious how many paragraphs you need to read down before you know when the 2.3.15 change stops.

Reply:
what about this

Reply:
well, they are all in one paragraph now.. so..

Reply:
But my idea is that the versionchanged would also be in the same paragraph.

Reply:
that renders ugly

Reply:
also we do that nowhere else.

Reply:
The main issue I have with these is that when you have text like:

```
blah1

version added 2.3.15

blah2
```

Then it's not very clear if the versionadded applies to blah1 or blah2 (or both!) This used to be done randomly in both ways until I changed it everywhere. One way I had fixed it was to do:

```
blah blah text

   .. versionadded:: 1.2.3
```

so the versionadded was indented after the text it applies to. I guess usually though it's done so that the versionadded is after a title and it applies to the whole section.